File: WillMoveToOwnRepo/ProjectorRays/ProjectorRays.DotNet/director/Sound.cs
using ProjectorRays.Common;

namespace ProjectorRays.Director;

public static class Sound
{
    public static int DecompressSnd(ReadStream input, WriteStream output, int castId)
    {
        // TODO: implement MP3 decoding similar to the original C++ version.
        // For now, simply copy the remaining bytes.
        if (input.Size == 0)
            return 0;

        int toCopy = (int)(input.Size - input.Pos);
        var data = input.ReadBytes(toCopy);
        output.WriteBytes(data);
        return toCopy;
    }
}

File: WillMoveToOwnRepo/ProjectorRays/src/ProjectorRays.Python/director/sound.py
from ..common.stream import ReadStream

class Sound:
    @staticmethod
    def decompress_snd(input_stream: ReadStream) -> bytes:
        if input_stream is None or input_stream.eof():
            return b""
        remaining = len(input_stream.data) - input_stream.pos
        return input_stream.read_bytes(remaining)
